pkg/operator/staticpod/controller/installer/installer_controller_test.go

```diff
@@ -766,8 +894,113 @@ func testSync(t *testing.T, firstInstallerBehaviour testSyncInstallerBehaviour)
 		t.Fatalf("expected current revision for node to be 1, got %d", revision)
 	}
 
-	t.Log("static pod is ready")
-	staticPod.Status.Conditions[0].Status = corev1.ConditionTrue
+	switch firstOperandBehaviour {
+	case testSyncOperandReady:
+		t.Log("static pod is ready")
+		staticPod.Status.Conditions[0].Status = corev1.ConditionTrue
+
+		if startupMonitorEnabled {
+			t.Log("startup-monitor notices static pod and update nodeStatus")
+			_, status, rv, err := fakeStaticPodOperatorClient.GetStaticPodOperatorState()
+			if err != nil {
+				t.Fatal(err)
+			}
+			status.NodeStatuses[0] = operatorv1.NodeStatus{
+				NodeName:        "test-node-1",
+				CurrentRevision: 3,
+			}
+			if _, err := fakeStaticPodOperatorClient.UpdateStaticPodOperatorStatus(rv, status); err != nil {
+				t.Fatal(err)
+			}
+		}
+
+	case testSyncOperandFallback:
+		t.Log("static pod is replaced with fallback by startup-monitor")
+
+		staticFallbackPod := staticPod.DeepCopy()
+		staticFallbackPod.Labels["revision"] = "1"
+		staticFallbackPod.Annotations[annotations.FallbackForRevision] = "3"
+		staticFallbackPod.Annotations[annotations.FallbackReason] = "ChashLooping"
+		staticFallbackPod.Annotations[annotations.FallbackMessage] = "pod is crashlooping"
+		staticFallbackPod.Status.Conditions[0].Status = corev1.ConditionTrue
+
+		kubeClient.PrependReactor("get", "pods", getPodsReactor(staticFallbackPod))
+
+		if err := c.Sync(context.TODO(), factory.NewSyncContext("InstallerController", eventRecorder)); err != nil {
+			t.Fatal(err)
+		}
+		_, currStatus, _, _ = fakeStaticPodOperatorClient.GetStaticPodOperatorState()
+		if currentRevision := currStatus.NodeStatuses[0].CurrentRevision; currentRevision != 1 {
+			t.Fatalf("expected current revision for node to be 1, got %d", currentRevision)
+		}
+		if count := currStatus.NodeStatuses[0].LastFailedCount; count != 1 {
+			t.Fatalf("expected fail count for node to be 1, got %d", count)
+		}
+
+		t.Log("not yet expecting second installer due to 1 min fallback backoff")
+		withCustomFallbackBackOff(10*time.Minute, func() {
+			if err := c.Sync(context.TODO(), factory.NewSyncContext("InstallerController", eventRecorder)); err != nil {
+				t.Fatal(err)
+			}
+			if len(installerPods) == 2 {
+				t.Fatal("not yet expected 2nd installer pod immediately, only after 1s")
+			}
+		})
```

Comment:
does this test take 10 mins?

is this really 1 second?
```
if len(installerPods) == 2 {
				t.Fatal("not yet expected 2nd installer pod immediately, only after 1s")
			}
```

Reply:
it returns immediately, but does not make progress. In another step I don't set 10min delay and I observe progress.